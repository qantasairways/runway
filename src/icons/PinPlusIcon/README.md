```js
<PinPlusIcon height="50" width="50" color="#000" />
```
