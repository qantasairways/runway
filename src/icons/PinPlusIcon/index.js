import React from 'react';
import PropTypes from 'prop-types';
import { generate } from 'shortid';

function PinPlusIcon({ height, width, className }) {
  const pathId = generate();
  const maskId = generate();

  return (
    <svg
      width={width}
      height={height}
      className={className}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <defs>
        <path
          id={pathId}
          d="M16.667 11.333a5.333 5.333 0 1 1 0 10.667 5.333 5.333 0 0 1 0-10.667zm2.666 6a.667.667 0 1 0 0-1.333h-2v-2A.667.667 0 1 0 16 14v2h-2a.667.667 0 0 0 0 1.333h2v2a.667.667 0 1 0 1.333 0v-2h2zm-9.366-.666c0 .893.182 1.778.533 2.6-1.033 1.266-1.833 2.066-1.833 2.066S2 14.667 2 8.667a6.667 6.667 0 1 1 13.333 0 9.167 9.167 0 0 1-.133 1.5 6.667 6.667 0 0 0-5.233 6.5zm-1.3-6a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
        />
      </defs>
      <g fillRule="evenodd">
        <mask id={maskId} fill="#fff">
          <use xlinkHref={`#${pathId}`} />
        </mask>
        <g mask={`url(#${maskId})`}>
          <path d="M0 0h24v24H0z" />
        </g>
      </g>
    </svg>
  );
}

PinPlusIcon.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
  className: PropTypes.string
};

PinPlusIcon.defaultProps = {
  height: '24',
  width: '24',
  className: ''
};

export default PinPlusIcon;
