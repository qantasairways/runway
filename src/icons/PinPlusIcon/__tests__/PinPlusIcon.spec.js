import React from 'react';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import PinPlusIcon from '..';

jest.mock('shortid');

describe('PinPlusIcon', () => {
  it('should render', () => {
    shortid.generate.mockReturnValue('1a2b3c');
    const component = shallow(
      <PinPlusIcon width="36px" height="36px" className="my-favourite-icon" />
    );

    expect(component).toMatchSnapshot();
  });

  it('should set the width', () => {
    const component = shallow(<PinPlusIcon width="36px" />);
    const result = component.find('svg').props().width;

    expect(result).toEqual('36px');
  });

  it('should set the height', () => {
    const component = shallow(<PinPlusIcon height="36px" />);
    const result = component.find('svg').props().height;

    expect(result).toEqual('36px');
  });

  it('should set the class name', () => {
    const component = shallow(<PinPlusIcon className="my-favourite-icon" />);
    const result = component.find('svg').props().className;

    expect(result).toEqual('my-favourite-icon');
  });
});
