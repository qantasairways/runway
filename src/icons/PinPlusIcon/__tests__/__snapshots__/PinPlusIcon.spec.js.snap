// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinPlusIcon should render 1`] = `
<svg
  className="my-favourite-icon"
  focusable="false"
  height="36px"
  viewBox="0 0 24 24"
  width="36px"
>
  <defs>
    <path
      d="M16.667 11.333a5.333 5.333 0 1 1 0 10.667 5.333 5.333 0 0 1 0-10.667zm2.666 6a.667.667 0 1 0 0-1.333h-2v-2A.667.667 0 1 0 16 14v2h-2a.667.667 0 0 0 0 1.333h2v2a.667.667 0 1 0 1.333 0v-2h2zm-9.366-.666c0 .893.182 1.778.533 2.6-1.033 1.266-1.833 2.066-1.833 2.066S2 14.667 2 8.667a6.667 6.667 0 1 1 13.333 0 9.167 9.167 0 0 1-.133 1.5 6.667 6.667 0 0 0-5.233 6.5zm-1.3-6a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
      id="1a2b3c"
    />
  </defs>
  <g
    fillRule="evenodd"
  >
    <mask
      fill="#fff"
      id="1a2b3c"
    >
      <use
        xlinkHref="#1a2b3c"
      />
    </mask>
    <g
      mask="url(#1a2b3c)"
    >
      <path
        d="M0 0h24v24H0z"
      />
    </g>
  </g>
</svg>
`;
